CMS-Enterprise/mint-app
# [NOREF] Dataloader sort and error utilities

## Description
Building off the success of generic relational sorting in EASI, this seeks to provide generic functionality to sort one to one, and one to many records. It also provides helper methods for returning an error for each result for a data loader
1.  Easi's one to one, and one to many funcs
2. Function based approximations of both functions
3. dataloader specific wrapper methods to handle conversion of standard results to a dataloader result
4. errorPerEachKey method for returning errors before dataloader results are generated



## How to test this change

1.  Ensure tests pass
2. If desired, hit model plan, basics, or operational solutions endpoints
3. verify the new sort functionality works as anticipated

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
